File: Sources/RTLSCore/LocationStore.swift
import Foundation

public struct PendingStats: Sendable, Equatable {
    public var count: Int
    public var oldestRecordedAt: Date?

    public init(count: Int, oldestRecordedAt: Date?) {
        self.count = count
        self.oldestRecordedAt = oldestRecordedAt
    }
}

public protocol LocationStore: Sendable {
    func insert(points: [LocationPoint]) async throws

    func fetchPendingPoints(limit: Int) async throws -> [LocationPoint]
    func pendingCount() async throws -> Int
    func oldestPendingRecordedAt() async throws -> Date?

    /// Single query returning both count and oldest timestamp — avoids 2 round-trips in hot path.
    func pendingStats() async throws -> PendingStats

    func markSent(pointIds: [UUID], sentAt: Date) async throws
    func markFailed(pointIds: [UUID], errorMessage: String) async throws
}

File: rtls-kmp/src/commonMain/kotlin/com/rtls/kmp/LocationStore.kt
package com.rtls.kmp

data class PendingStats(val count: Int, val oldestRecordedAtMs: Long?)

interface LocationStore {
    suspend fun insert(points: List<LocationPoint>)
    suspend fun fetchPendingPoints(limit: Int): List<LocationPoint>
    suspend fun pendingCount(): Int
    suspend fun oldestPendingRecordedAt(): Long?
    /** Single query returning both count and oldest timestamp — avoids 2 round-trips in hot path. */
    suspend fun pendingStats(): PendingStats = PendingStats(pendingCount(), oldestPendingRecordedAt())
    suspend fun markSent(pointIds: List<String>, sentAtMs: Long)
    suspend fun markFailed(pointIds: List<String>, errorMessage: String)
}
